spec/pushmi_pullyu/aip/downloader_spec.rb

```diff
@@ -3,37 +3,106 @@
+     '3cbc75f9-7bad-4d62-962e-911f911bd70e']
+  end
+  let(:aip_folder) { "#{workdir}/#{uuid}" }
+  let(:downloader) { PushmiPullyu::AIP::Downloader.new(entity_definition, aip_folder) }
+  let(:folders) do
+    [
+      "tmp/downloader_spec/#{uuid}/data",
+      "tmp/downloader_spec/#{uuid}/data/logs",
+      "tmp/downloader_spec/#{uuid}/data/logs/files_logs",
+      "tmp/downloader_spec/#{uuid}/data/logs/files_logs/#{file_set_uuids[0]}",
+      "tmp/downloader_spec/#{uuid}/data/logs/files_logs/#{file_set_uuids[1]}",
+      "tmp/downloader_spec/#{uuid}/data/objects",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[0]}",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[1]}",
+      "tmp/downloader_spec/#{uuid}/data/objects/files",
+      "tmp/downloader_spec/#{uuid}/data/objects/files/#{file_set_uuids[0]}",
+      "tmp/downloader_spec/#{uuid}/data/objects/files/#{file_set_uuids[1]}"
+    ]
+  end
+  let(:files_copied) do
+    [
+      "tmp/downloader_spec/#{uuid}/data/objects/files/#{file_set_uuids[0]}/theses.jpg",
+      "tmp/downloader_spec/#{uuid}/data/objects/files/#{file_set_uuids[1]}/image-sample.jpeg"
+    ]
+  end
+  let(:files_downloaded) do
+    [
+      # aipcreation.log is not downladed but it makes sense to add it here
+      "tmp/downloader_spec/#{uuid}/data/logs/aipcreation.log",
+      "tmp/downloader_spec/#{uuid}/data/logs/files_logs/#{file_set_uuids[0]}/content_fixity_report.n3",
+      "tmp/downloader_spec/#{uuid}/data/logs/files_logs/#{file_set_uuids[1]}/content_fixity_report.n3",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/object_metadata.n3",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/file_order.xml",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[0]}/file_set_metadata.n3",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[0]}/"\
+      'original_file_metadata.n3',
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[1]}/file_set_metadata.n3",
+      "tmp/downloader_spec/#{uuid}/data/objects/metadata/files_metadata/#{file_set_uuids[1]}/"\
+      'original_file_metadata.n3'
+    ]
   end
-  let(:aip_folder) { "#{workdir}/#{noid}" }
-  let(:downloader) { PushmiPullyu::AIP::Downloader.new(noid, aip_folder) }
 
   before do
     allow(PushmiPullyu.logger).to receive(:info)
     allow(PushmiPullyu.logger).to receive(:debug)
     allow(PushmiPullyu).to receive(:options) { options }
     FileUtils.mkdir_p(workdir)
-    allow(PushmiPullyu::AIP::User)
-      .to receive(:find).with(2705).and_return(OpenStruct.new(email: 'admin@example.com'))
   end
 
   after do
     FileUtils.rm_rf(workdir)
   end
 
   describe '#run' do
+    it 'copies the correct files' do
```

Comment:
I need to fix all tests dealing with file copying since the response from the local instance of Jupiter gives an absolute file that does not exist when the vcr cassette is played elsewhere. I plan to fix this by filtering the response from Jupiter and changing it to a path relative to the project. Any other suggestions for this are welcome.

Reply:
I will take a look at that configuration, I was having problems with it connecting to my local instance of jupiter but maybe I was using it wrong. Will check again.